server/storage.lua

```diff
@@ -93,18 +94,30 @@ end
 
 ---@param citizenId string
 ---@param license string
----@param plate string
-function UpdateVehicleEntityOwner(citizenId, license, plate)
-    MySQL.update('UPDATE player_vehicles SET citizenid = ?, license = ? WHERE plate = ?', {citizenId, license, plate})
+---@param vehicleId integer
+function UpdateVehicleEntityOwner(citizenId, license, vehicleId)
+    MySQL.update('UPDATE player_vehicles SET citizenid = ?, license = ? WHERE id = ?', {citizenId, license, vehicleId})
+end
+
+---@param id integer
+---@return VehicleFinancingEntity
+function FetchFinancedVehicleEntityById(id)
+    return MySQL.single.await('SELECT * FROM vehicle_financing WHERE vehicleId = ? AND balance > 0 AND financetime < 1', {id})
 end
 
 ---@param citizenId string
----@return VehicleEntity[]
+---@return VehicleFinancingEntity
 function FetchFinancedVehicleEntitiesByCitizenId(citizenId)
-    return MySQL.query.await('SELECT * FROM player_vehicles WHERE citizenid = ? AND balance > 0 AND financetime < 1', {citizenId})
+    return MySQL.query.await('SELECT vehicle_financing.* FROM vehicle_financing INNER JOIN player_vehicles ON player_vehicles.citizenid = "KLH31301" WHERE vehicle_financing.vehicleId = player_vehicles.id AND vehicle_financing.balance > 0 AND vehicle_financing.financetime > 1', {citizenId})
 end
 
----@param plate string
-function DeleteVehicleEntity(plate)
-    MySQL.query('DELETE FROM player_vehicles WHERE plate = ?', {plate})
+---@param license string
+---@return VehicleFinancingEntity
+function FetchFinancedVehicleEntitiesByLicense(license)
+    return MySQL.query.await('SELECT * FROM vehicle_financing AS vf INNER JOIN players AS p ON p.citizenid = ? INNER JOIN player_vehicles AS pv ON pv.citizenid = p.citizenid AND vf.balance > 0 AND vf.financetime < 1', {license})
```

Comment:
What columns do we take from the player table?

Reply:
citizenId

Reply:
Consider limiting the selection to avoid unnecessary columns reads

Reply:
Already did.